### PR TITLE
bugfixing deprecated functions for clang 11

### PR DESCRIPTION
**Pull request by baykaner**

### CODE REVIEW

File: libs/storage/tests/gtest/key_value_index_tests.cpp [hunk 1]
```diff
@@ -194,7 +195,9 @@ bool RandomInsertHashConsistency()
   // REDO
   key_index.New("test1.db");
 
-  std::random_shuffle(values.begin(), values.end());
+  std::random_device rd;
```
Comment (n-hutton) [inline libs/storage/tests/gtest/key_value_index_tests.cpp:198]: why is it complaining about this?

File: libs/core/include/core/string/trim.hpp [hunk 2]
```diff
@@ -29,15 +29,13 @@ namespace string {
 
 inline void TrimFromRight(std::string &s)
 {
-  s.erase(s.begin(),
-          std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+  s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int c) { return !std::isspace(c); }));
```
Comment (n-hutton) [inline libs/core/include/core/string/trim.hpp:32]: is this related?

File: libs/network/include/network/management/abstract_connection_register.hpp [hunk 3]
```diff
@@ -48,9 +48,9 @@ class AbstractConnectionRegister : public std::enable_shared_from_this<AbstractC
   {}
 
   AbstractConnectionRegister(AbstractConnectionRegister const &other) = delete;
-  AbstractConnectionRegister(AbstractConnectionRegister &&other)      = default;
+  AbstractConnectionRegister(AbstractConnectionRegister &&other)      = delete;
```
Comment (n-hutton) [inline libs/network/include/network/management/abstract_connection_register.hpp:51]: 👍

File: libs/storage/tests/gtest/file_object/file_object_tests.cpp [hunk 4]
```diff
@@ -195,7 +196,9 @@ TEST_F(FileObjectTests, ResizeAndWriteFiles)
 
   for (std::size_t i = 0; i < 10; ++i)
   {
-    std::random_shuffle(consistency_check_.begin(), consistency_check_.end());
+    std::random_device rd;
```
Comment (n-hutton) [inline libs/storage/tests/gtest/file_object/file_object_tests.cpp]: I suspect that constructing this is expensive and not something that you want to do inside the loop

Comment (baykaner) [reply libs/storage/tests/gtest/file_object/file_object_tests.cpp]: I agree but it's a test, and presumably it's what was happening already inside random_shuffle?

File: libs/core/include/core/string/trim.hpp (hunk 2 above)
Comment (baykaner) [reply libs/core/include/core/string/trim.hpp:32]: ptr_fun deprecated. lambda necessary

File: libs/storage/tests/gtest/file_object/file_object_tests.cpp (hunk 4 above)
Comment (baykaner) [reply libs/storage/tests/gtest/file_object/file_object_tests.cpp]: anyway, fixed

**Review by n-hutton — APPROVED** [review]

File: libs/storage/tests/gtest/key_value_index_tests.cpp (hunk 1 above)
Comment (baykaner) [reply libs/storage/tests/gtest/key_value_index_tests.cpp:198]: random shuffle is deprecated
